apps/examples/iperf2/compat/Thread.c

```diff
@@ -0,0 +1,417 @@
+#if   defined( HAVE_WIN32_THREAD )
+DWORD WINAPI
+#else
+void*
+#endif
+thread_run_wrapper( void* paramPtr ) {
+    struct thread_Settings* thread = (struct thread_Settings*) paramPtr;
+
+    // which type of object are we
+    switch ( thread->mThreadMode ) {
+        case kMode_Server:
+            {
+                /* Spawn a Server thread with these settings */
+                server_spawn( thread );
+            } break;
+        case kMode_Client:
+            {
+                /* Spawn a Client thread with these settings */
+                client_spawn( thread );
+            } break;
+        case kMode_Reporter:
+            {
+                /* Spawn a Reporter thread with these settings */
+                reporter_spawn( thread );
+            } break;
+        case kMode_Listener:
+            {
+                // Increment the non-terminating thread count
+                thread_register_nonterm();
+                /* Spawn a Listener thread with these settings */
+                listener_spawn( thread );
+                // Decrement the non-terminating thread count
+                thread_unregister_nonterm();
+            } break;
+        default:
+            {
+                FAIL(1, "Unknown Thread Type!\n", thread);
+            } break;
+    }
+
+#ifdef HAVE_POSIX_THREAD
+    // detach Thread. If someone already joined it will not do anything
+    // If noone has then it will free resources upon return from this
+    // function (Run_Wrapper)
+    pthread_detach(thread->mTID);
+#endif
+
+    // decrement thread count and send condition signal
+    Condition_Lock( thread_sNum_cond );
+    thread_sNum--;
+    Condition_Signal( &thread_sNum_cond );
+    Condition_Unlock( thread_sNum_cond );
+
+    // Check if we need to start up a thread after executing this one
+    if ( thread->runNext != NULL ) {
+        thread_start( thread->runNext );
+    }
+
+    // Destroy this thread object
+    Settings_Destroy( thread );
+
+    return 0;
+} // end run_wrapper
+
+/* -------------------------------------------------------------------
+ * Wait for all thread object's execution to complete. Depends on the
+ * thread count being accurate and the threads sending a condition
+ * signal when they terminate.
+ * ------------------------------------------------------------------- */
+void thread_joinall( void ) {
+    Condition_Lock( thread_sNum_cond );
+    while ( thread_sNum > 0 ) {
+        Condition_Wait( &thread_sNum_cond );
+    }
+    Condition_Unlock( thread_sNum_cond );
+} // end Joinall
+
+
+/* -------------------------------------------------------------------
+ * Compare the thread ID's (inLeft == inRight); return true if they
+ * are equal. On some OS's nthread_t is a struct so == will not work.
+ * TODO use pthread_equal. Any Win32 equivalent??
+ * ------------------------------------------------------------------- */
+int thread_equalid( nthread_t inLeft, nthread_t inRight ) {
```

Comment:
same as above & please check this for entire file

Reply:
it's opensource so It would be better not to apply coding rule